Xamarin.Forms.Platform.Android/Extensions/ImageViewExtensions.cs

```diff
@@ -103,6 +107,16 @@ bool SourceIsNotChanged(IImageElement imageElement, ImageSource imageSource)
 			}
 		}
 
+		public static async Task SetImagePlaceholder(this AImageView imageView, ImageSource placeholder)
+		{
+			using (var drawable = await imageView.Context.GetFormsDrawableAsync(placeholder))
+			{
+				// only set the image if we are still on the same one
+				if (!imageView.IsDisposed())
```

Comment:
This if statement will also need this check `SourceIsNotChanged(newView, newImageSource)`

Maybe move this check `!imageView.IsDisposed() && SourceIsNotChanged(newView, newImageSource)`

into an extension method? `ShouldStillSetImage`

And then reuse that both places?

Reply:
Placeholder here still needs to check if the source has changed and view hasn't been disposed